hack/gc-host-local.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+CLEAN=false
+DATA_DIR="/run/antrea/cni/networks/antrea/"
+BACKUP_DIR="${DATA_DIR}leaked/"
+
+_usage="Usage: $0 [--clean]
+
+Clean up leaked IPs from host-local IPAM plugin used by Antrea. crictl tool and proper permissions is required.
+
+        --clean                  Clean mode. If not set, this script will only print the leaked IPs and won't remove them."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --clean)
+    CLEAN=true
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+LEAKED_IPS=()
+
+function generate_leaked_ips() {
+    CONTAINER_IDS=( $(crictl pods | awk '{print $1}') )
+    unset CONTAINER_IDS[0]
+    ALLOCATED_IPS=()
+    for IPV4 in `ls ${DATA_DIR}*.*.*.*`; do
+        ALLOCATED_IPS+=(${IPV4})
+    done
+    for IPV6 in `ls ${DATA_DIR}*:*:*`; do
+        ALLOCATED_IPS+=(${IPV6})
+    done
+    for IP in "${ALLOCATED_IPS[@]}"; do
```

Comment:
this order may remove normal IPs unexpectedly, considering this case:
1. Get `CONTAINER_IDS`
2. Pod A is created
3. `ls ${DATA_DIR}` returns IP of Pod A
4. It doesn't find Pod A's containerID in `CONTAINER_IDS` and removes it mistakenly.

Instead, it should get IPs first, then `CONTAINER_IDS`.

And perhaps the script should acquire `/run/antrea/cni/networks/antrea/lock` first to avoid all race conditions that could happen.

Reply:
Fixed the sequence and added flock to lock the file.